### ISSUE TITLE
Add arithmetic bracketing sniff ?

### ISSUE THREAD

**Issue by jrfnl** [WordPress-Coding-Standards/WordPress-Coding-Standards#589]
As operator precedence is a mystery to a lot of programmers, demanding brackets around arithmetics is good practice and prevents bugs.

So, should we add it ? `Squiz.Formatting.OperatorBracket.MissingBrackets` does the trick.

Optionally this sniff could be forked to throw a warning instead of an error.

As it's not in the handbook, should probably go in `extra`.

Opinions ?

**Comment by westonruter** [comment]
I suppose it's not a bad idea for WordPress-Extra.

**Comment by JDGrimes** [comment]
IMO, this seems silly. I understand flagging occasions when multiple operators are being used without any parentheses, but flagging `$a + $b` seems ridiculous. I'm pretty sure I'd disable this in my custom config if it was added to `extra`. That doesn't mean that we can't add it, but I'd want the sniff to be more judicious about what it flags before I'd use it, personally.

**Comment by jrfnl** [comment]
Ok, I suggested it as code like the below (ok, well, without the `var_dump()`'s) is often enough found in WP core and the WP ecosystem at large...

```php
$a = '200';
$b = false;
$c = 15;
	
$x = (bool) $a + $b;
$y = ! $a + $c;
	
var_dump( $x ); // what will this be ? boolean true or int 200 or int 1 ?
var_dump( $y ); // what will this be ? boolean false or int 215 or or 15 ?
```

**Comment by JDGrimes** [comment]
> $x = (bool) $a + $b;
> $y = ! $a + $c;

This is something that I'd be happy with the sniff flagging, as it is using multiple operators. It just seems silly to put parentheses in `$c = $a + $b`.

**Comment by jrfnl** [comment]
@JDGrimes That would require a custom sniff I think. The sniff suggested is the best I could find within the existing sniffs in PHPCS, but maybe someone else's search skills are better ?

It's also the reason to suggest this sniff for `extra`. If we'd have a better sniff (arithmetics with multiple orperators), I would advocate it should go into the `core` ruleset as that will be a bug 95% of the time.

**Comment by GaryJones** [comment]
`Core` is meant to be reserved for those things that are mentioned in the Handbook. Sure, it would be great if a few things from our Extra got into the Handbook as a "best practice", but that's a different matter (and people to convince) entirely.
